samza-core/src/main/java/org/apache/samza/runtime/ContainerLaunchUtil.java

```diff
@@ -147,6 +149,14 @@ private static void run(
         heartbeatMonitor.start();
       }
 
+      if (new JobConfig(config).getApplicationMasterHighAvailabilityEnabled()) {
+        ExecutionContainerIdManager executionContainerIdManager = new ExecutionContainerIdManager(
+            new NamespaceAwareCoordinatorStreamStore(coordinatorStreamStore, SetExecutionContainerIdMapping.TYPE));
+        if (executionContainerIdManager != null && execEnvContainerId.isPresent()) {
```

Comment:
we just create the manager above. why is this null check? should be `execEnvContainerId` instead. 

I still think you don't need this check. You can use `Optional::ifPresent` and write it.